### PR TITLE
Connor #46

### LINKED ISSUE #46: If one website fails then we shouldn't return only an error for everything

### PR DESCRIPTION
**Pull request by connorjoleary**
The biggest change with this is that the response object is now larger and more info is saved. Also you can get the results of a previous run by providing the id.

### CODE REVIEW

File: backend/aws/lambda/app.py [hunk 1]
```diff
@@ -76,31 +94,37 @@ def lambda_handler(event, context):
...
             payload = load_payload(event)
-            response = respond(None, operations[operation](payload))
+            response = operations[operation](payload, conn)
         except Exception as e:
+            traceback.print_tb(e.__traceback__)
             print(e)
+            response = e
     else:
-        response = respond(ValueError('Unsupported method "{}"'.format(operation)))
+        response = Exception('Unsupported method "{}"'.format(operation))
```
Comment (deadmau6) [inline backend/aws/lambda/app.py]: I think `f-string` is the new standard?

File: backend/aws/lambda/app.py [hunk 2]
```diff
@@ -17,6 +20,7 @@
     service_name='secretsmanager',
     region_name=config['secret']['region']
 )
+new_submission = True
```
Comment (deadmau6) [inline backend/aws/lambda/app.py:23]: It seems like a few of these methods like `grab_response` and/or `call_deepcite` could be abstracted into a single class and then `new_submission` could be a property. It could be like an `Operations class` or `ResponseHandler class`. But maybe that also should be done in a separate PR.

File: backend/aws/lambda/app.py [hunk 3]
```diff
@@ -76,31 +94,37 @@ def lambda_handler(event, context):
...
-    else:
-        print("SUCCESS: Connection to RDS instance succeeded")
 
-    print(response)
-        
-    return response
+    return respond(payload.get('response_size', 'small'), response)
 
-# print(lambda_handler({"requestContext": {"http": {"method": "GET"}}},0))
+# body = "{\"id\": \"fdaeasfsd\", \"resonse_size\": \"small\", \"claim\":\"the death of Sherlock Holmes almost destroyed the magazine thries. When Arthur Conan Doyle killed him off in 1893, 20,000 people cancelled their subscriptions. The magazine barely survived. Its staff referred to Holmes’ death as “the dreadful event”.\", \"link\":\"http://www.bbc.com/culture/story/20160106-how-sherlock-holmes-changed-the-world\"}"
```
Comment (deadmau6) [inline backend/aws/lambda/app.py:129]: I would be ok with having some of these as like `example.json` files especially if you think that would be useful for testing?

File: backend/aws/lambda/create_response.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+def trim_response(error, results):
+    if len(results) == 0:
+        raise('No results returned, should include at least source')
+    
+    shortened_results = [{'link': result['link'], 'source': result['source'], 'score': result['score']} for result in results]
+    source = shortened_results[0]
+    shortened_results = shortened_results[1:]
+    sorted_results = sorted(shortened_results, key=lambda k: k['score'], reverse=True)
+
+    return json.dumps([source]+sorted_results[:3])
```
Comment (deadmau6) [inline backend/aws/lambda/create_response.py]: Why `:3`? Is this top 3?

File: backend/claim.py [hunk 5]
```diff
@@ -1,65 +1,234 @@
...
+from wiki_scraper import wiki
 import sys
 import io
+import re
+import os
+import queue as q
+from config import config
+import uuid
+
+CWD_FOLDER = os.path.dirname(os.path.abspath(__file__))
```
Comment (deadmau6) [inline backend/claim.py:15]: This should be part of `config`

File: backend/aws/lambda/app.py (hunk 2 above)
Comment (connorjoleary) [reply backend/aws/lambda/app.py:23]: I can agree. I have some code that puts the DB calls in another class, but this would be good too

File: backend/aws/lambda/create_response.py (hunk 4 above)
Comment (connorjoleary) [reply backend/aws/lambda/create_response.py]: Yah, that is what that syntax means, but it should be a config value, not just hard coded top 3

File: backend/claim.py (hunk 5 above)
Comment (connorjoleary) [reply backend/claim.py:15]: I don't actually think this one should, because this just adds a file path relative to this file.

Comment (connorjoleary) [reply backend/claim.py:15]: I don't actually think this one should, because this just adds a file path relative to this file.
